0.1.0/lib.typ

```diff
@@ -0,0 +1,159 @@
+#import "@preview/acrostiche:0.3.1": init-acronyms, print-index
+#import "lib/research-questions.typ": init-rqs
+
+// workaround for lack of std scope
+#let std-biblio = bibliography
+
+// store in states such that these need to be passed
+// only once, in the `thesis` function
+#let _authors = state("authors", ())
+#let _title = state("title", [])
+#let _supervisors = state("supervisors", ())
+#let _year = state("year", [2023-2024])
+
+#let thesis(
+  // The title of this thesis
+  title: [The title of the thesis],
+  // the authors of this thesis
+  authors: ("John Doe",),
+  // The supervisors for this thesis
+  supervisors: ("Prof. Dr. Jane Doe",),
+  // the academic year
+  year: [2023-2024],
+  // a `#bibliography(...)` or `none`
+  bibliography: none,
+  // the actual content of the thesis
+  body
+) = {
+  set document(
+    title: title, 
+    author: authors,
+  )
+  // set states to the given parameters
+  _authors.update(authors)
+  _title.update(title)
+  _supervisors.update(supervisors)
+  _year.update(year)
+
+
+  set page(
+    paper: "a4",
+    margin: 2cm,
+    header: context {
+      let elems = query(
+        selector(heading).before(here()))
+
+      let headings_at_this_page = query(
+        heading.where(level: 1)
+      ).find(h => h.location().page() == here().page())
+
+      let page_has_no_heading = headings_at_this_page == none 
+
+      if elems.len() != 0 and page_has_no_heading {
+        let body = elems.last().body
+        align(right, emph(body))
+      }
+    },
+    numbering: "I"
```

Comment:
The numbering is set to roman numerals here, but never set back to normal. Or am i missing something?